stripe/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -189,9 +190,12 @@ internal class DefaultFlowController internal constructor(
         val config = initData.config
         val paymentSelection = viewModel.paymentSelection
         if (paymentSelection == PaymentSelection.GooglePay) {
+            if (initData.stripeIntent !is PaymentIntent) {
+                error("Google Pay currently supported only for PaymentIntents")
```

Comment:
Throwing exceptions from this function isn't new, but it probably would be good to document in the api.

Reply:
This is one of those "should never happen" cases, since we only show Google Pay as an option for PaymentIntent. Still I think this check is good to be explicit about the error and make it easier to debug.

Reply:
I agree, seems like our api for confirm should say that we throw exceptions, no?